src/Extensions.cs

```diff
@@ -327,7 +327,7 @@ public static void AddImports(this IModelType type, HashSet<string> imports)
         public static bool ShouldBeSyntheticType(this IModelType type)
         {            
             return (type is PrimaryType || type is SequenceType || type is DictionaryType || type is EnumType || 
-                (type is CompositeTypeGo && (type as CompositeTypeGo).IsPolymorphicResponse()));
+                (type is CompositeType && (type as CompositeTypeGo).IsPolymorphicResponse()));

```

Comment:
How do we end up with an object that isn't a CompositeTypeGo?  That seems like a bug.

Reply:
Well, this func is called on CompositeType when it has not been casted to a CompositeTypeGo.